src/watch.rs

```diff
@@ -101,10 +101,14 @@ impl WatchSystem {
         }
     }
 
-    fn update_ignore_list(&mut self, path: PathBuf) {
-        if !self.ignored_paths.contains(&path) {
-            self.ignored_paths.push(path);
+    fn update_ignore_list(&mut self, path: PathBuf) -> Result<()> {
+        let canon_path = path
+            .canonicalize()
+            .with_context(|| format!("Could not canonicalize path: {}", path.to_string_lossy()))?;
+        if !self.ignored_paths.contains(&canon_path) {
+            self.ignored_paths.push(canon_path)
         }
+        Ok(())
```

Comment:
BTW @granitrocky I've updated the code in master to force the callsite to canonicalize the path instead of doing it here.

I do want to get rid of this abstraction ultimately ... but that will requite a bit of refactoring. So for now, we should be g2g and hopefully the documentation I've added will help to guard against regressions.